docs/plugins/reindex.asciidoc

```diff
@@ -0,0 +1,608 @@
+[[plugins-reindex]]
+=== Reindex Plugin
+
+The reindex plugin adds two APIs:
+
+* `_update_by_query` updates all documents matching a query in place.
+* `_reindex` copies documents from one index to another.
+
+These APIs are brothers so they live in the same plugin. Both use
+{ref}/search-request-scroll.html[Scroll] and {ref}/docs-bulk.html[Bulk] APIs
+to send an index request per document though they may not work that way in the
+future. Regardless of how they work internally they will always work **as**
+**though** they send an index request per document that matches a scroll.
```

Comment:
I wrote it this way because I think we might be able to optimize certain types of these requests and not others. The scroll based implementation here is always going to live as a "reference" implementation but there might be more specialized implementations, like if you don't change the mapping at all then `_reindex` might be able to get away with doing filesystem level things.